models/user.model.js

```diff
@@ -26,21 +26,24 @@ const userSchema = new Schema({
   password: {
     type: String,
     require: [true, "password field is required"],
-    // validate: {
-    //   validator: function(v) {
-    //     return v.length > 6 && /\d+/.test(v)
-    //   },
-    //   message: props => `password must be 6 characters long and must contain a number`
-    // }
+    validate: {
+      validator: function(v) {
+        return v.length >= 6 && /\d+/.test(v)
+      },
+      message: props => `password must be 6 characters long and must contain a number`
+    }
   },
 
   cart: {
     type: Array,
   },
   wishlist: {
```

Comment:
Why are you doing it like this instead of giving ObjectId ref of product

Reply:
Ok I will fix this.